### PR TITLE
Overview updated 

### PR DESCRIPTION
**Pull request by poioper**
1. Converted overall layout from grid to flex.
2. Fixed the social media links UI.
3. Updated product detail UI.
4. Image carousel done.

### CODE REVIEW

File: client/src/components/App.jsx [hunk 1]
```diff
@@ -95,8 +95,11 @@ class App extends React.Component {
           <GlobalStyles/>
           <Container>
             <h1>Welcome to Atelier!</h1>
-            <Overview currentProductId={this.state.currentProductId}></Overview>
-            <div><Reviews currentProductId = '40344'currentProductRating = {this.state.currentProductRating}/></div>
```
Comment (codingavatar) [inline client/src/components/App.jsx:99]: Did you delete the Reviews component?

File: client/src/components/Overview/Cart.jsx [hunk 2]
```diff
@@ -1,56 +1,58 @@
...
-          }
-        </select>
-      </div>
-      <div className="custom-select" style={{'width': '200px'}}>
-        <select defaultValue={{label: 'SELECT QUANTITY:', value: 0}}>
-          {
-            // TODO: Don't know why, chrome warning option needs key
-            selectedSizeQuantity.map(item => {
-              return item.key <= 15 && <option key={item.key} value={item.key}>{item.key}</option>;
+      <CustomSelect defaultValue={0} onChange={handleChange}>
```
Comment (codingavatar) [inline client/src/components/Overview/Cart.jsx:23]: You probably need to pass in the event to the handleChange invocation
ie: onChange={(event) => handleChange(event)}

File: client/src/components/Overview/ImageCarousel.jsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+import { StyledImageCarousel } from '../styles/Overview/ImageCarousel.styled';
+import ImageItem from './ImageItem';
+import $ from 'jquery';
+
+/**
+ * ImageCarousel that holds all the thumbnails in the default view
+ */
+const ImageCarousel = ({productStyle, thumbnailChange, selectedThumbnailIndex}) => {
+
+  // currentIndex is used to track where the boundary of the carousel-container is.
```
Comment (codingavatar) [inline client/src/components/Overview/ImageCarousel.jsx:11]: glad it worked!

File: client/src/components/Overview/ProductDetail.jsx [hunk 4]
```diff
@@ -1,30 +1,45 @@
 import { StyledProductDetail } from '../styles/Overview/ProductDetail.styled';
 const ProductDetail = ({description, slogan, features}) => {
+
+  const shareToSocialMedia = (socialMedia) => {
+    alert(`Shared to ${socialMedia}`);
```
Comment (codingavatar) [inline client/src/components/Overview/ProductDetail.jsx:5]: Is this supposed to do anything else with the social media?

File: client/src/components/styles/Overview/ImageShowcase.styled.js [hunk 5]
```diff
@@ -1,20 +1,121 @@
...
+  .slideshowSlider {
+    white-space: nowrap;
+    transition: ease 1000ms;
+    display: flex;
+    align-items: center;
+
+    // If the user hovers over the main image
+    // the mouse icon should change to show a magnifying glass
+    // https://stackoverflow.com/questions/13171968/magnifying-glass-zoom-cursor-over-image
+    &: hover {
```
Comment (codingavatar) [inline client/src/components/styles/Overview/ImageShowcase.styled.js:23]: Nice touch!!

**Review by codingavatar — APPROVED** [review]
Looks great!

File: client/src/components/App.jsx (hunk 1 above)
Comment (poioper) [reply client/src/components/App.jsx:99]: Shouldn't, I only commented it out.

Comment (poioper) [reply client/src/components/App.jsx:99]: Yeah, it's in below section.

File: client/src/components/Overview/Cart.jsx (hunk 2 above)
Comment (poioper) [reply client/src/components/Overview/Cart.jsx:23]: Yeah, I will come back to this later.

File: client/src/components/Overview/ProductDetail.jsx (hunk 4 above)
Comment (poioper) [reply client/src/components/Overview/ProductDetail.jsx:5]: Doc says the following,
"buttons should provide the ability to share this product on popular social media platforms"
There are some libraries that can help to do this, I am not sure if we can use them or not. If not, I am not sure if we want to implement one ourselves.

File: client/src/components/App.jsx (hunk 1 above)
Comment (codingavatar) [reply client/src/components/App.jsx:99]: ah gotcha, I added it back in when I merged with mine!

File: client/src/components/Overview/ProductDetail.jsx (hunk 4 above)
Comment (codingavatar) [reply client/src/components/Overview/ProductDetail.jsx:5]: Hm yeah, might not be worth figuring out right now

**Pull request merged**